File: app/src/main/java/edu/gatech/donatrix/dao/LocationDao.java
package edu.gatech.donatrix.dao;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

import edu.gatech.donatrix.model.Location;
import edu.gatech.donatrix.model.User;

public class LocationDao {
    public static List<Location> getLocations(Context context) {
        if (Database.getInstance(context).getLocations().isEmpty()) {
            Database.getInstance(context).loadLocations(context);
        }
        return Database.getInstance(context).getLocations();
    }
    public static Location getLocationByID(Integer i, Context context) {
        return Database.getInstance(context).getLocationByID(i);
    }
    public static void addLocationEmployee(User user, Location location, Context context) {
        Database.getInstance(context).addLocationEmployee(user, location);
    }
    public static List<String> getLocationNames(Context context) {
        if (Database.getInstance(context).getLocations().isEmpty()) {
            Database.getInstance(context).loadLocations(context);
        }
        List<Location> locList = Database.getInstance(context).getLocations();
        List<String> strList = new ArrayList<>();
        for (Location loc: locList) {
            strList.add(loc.getName());
        }
        return strList;
    }
}

File: app/src/main/java/edu/gatech/donatrix/dao/LocationDao.kt
package edu.gatech.donatrix.dao

import android.content.Context

import java.util.ArrayList

import edu.gatech.donatrix.model.Location
import edu.gatech.donatrix.model.User

object LocationDao {
    fun getLocations(context: Context): List<Location> {
        if (Database.getInstance(context)!!.locations.isEmpty()) {
            Database.getInstance(context)!!.loadLocations(context)
        }
        return Database.getInstance(context)!!.locations
    }

    fun getLocationByID(i: Int?, context: Context): Location? {
        return Database.getInstance(context)!!.getLocationByID(i)
    }

    fun addLocationEmployee(user: User, location: Location, context: Context) {
        Database.getInstance(context)!!.addLocationEmployee(user, location)
    }

    fun getLocationNames(context: Context): List<String?> {
        if (Database.getInstance(context)!!.locations.isEmpty()) {
            Database.getInstance(context)!!.loadLocations(context)
        }
        val locList = Database.getInstance(context)!!.locations
        val strList = ArrayList<String?>()
        for (loc in locList) {
            strList.add(loc.name)
        }
        return strList
    }
}
